vllm/model_executor/layers/fused_moe/cutlass_moe.py

```diff
@@ -112,86 +108,142 @@ def apply(
             local_topk_ids = topk_ids
 
         topk = local_topk_ids.shape[1]
+        local_E = w1.shape[0]
+
+        if expert_num_tokens is not None:
+            non_zero_mask = expert_num_tokens[:] != 0
+            masked_local_E = int(non_zero_mask.sum().item())
```

Comment:
Is this going to interfere with cudagraphs?

Reply:
potentially... I think I can circumvent it with a custom CUDA kernel and extra mapping for `expert_offsets` and `problem_sizes1/2` if needed